### PR TITLE
US-6.2 Feature/Implement ChatContext Strategy (Refactor State)

### PR DESCRIPTION
**Pull request by TQT279**
### 🚀 Summary

This PR implements the core refactoring of the Chat System, transitioning from simple state management to a robust **Stateful Context Architecture** (`GLOBAL`, `PRIVATE`, `ROOM`). It also standardizes the connection **Lifecycle Management** to prevent memory leaks and ensure data consistency across registries.

BuildSuccess

#### 1. Architecture & Model

* **`UserSession`**: Refactored to be an **Immutable Object**. Added `ChatContext` field to track user state (currently `GLOBAL`, `PRIVATE`, `ROOM`).
* **`ChatContext`**: New Enum defining the three operational modes.

#### 2. Registry Layer (Data Store)

* **`UserRegistry`**:
* Decoupled "Physical" (Session ID) from "Logical" (Username) tracking.
* **Bugfix:** Fixed logic allowing active sessions to overwrite/rename usernames. Sessions must now logout before re-registering.
* Added `updateUser()` method to support immutable state updates (Context transitions).

* **`PrivateChatRegistry`**:
* Implemented **Reverse Index** (`target -> senders`) for **O(1)** lookup performance.
* Added automatic cleanup logic when users disconnect.

#### 3. Service Layer & Logic

* **`ChatService`**:
* Refactored `routeMessage` to use `switch(context)` for cleaner routing logic instead of nested if-else.
* **SRP Fix:** Removed side-effect cleanup logic (deleting users) from the Service. The Service is now **Pure Logic**.

* **`ChatWebSocketHandler`**:
* Now fully responsible for **Lifecycle Management**.
* Centralized cleanup calls (`removeUser`, `removeSession`, `leaveRoom`) in `afterConnectionClosed` to strictly prevent **Memory Leaks**.

#### 4. Commands

* **`SelectCommand`**: Updated to switch the User Context to `PRIVATE` upon execution.
* **`LeaveCommand`**: Updated to reset the User Context back to `GLOBAL` and clean up targets.

#### 5. Testing (Quality Assurance)

* **Unit Tests**: comprehensive updates to `ChatServiceTest` and `UserRegistryTest` to align with the new Context architecture.
* **Integration Tests**: Fixed `CommandIntegrationTest` to correctly simulate the session initialization flow (Physical -> Logical -> Register).
* **Coverage**: Verified edge cases including Race Conditions, Memory Leaks, and Context Switching.

ChatService
ChatWebSocketHandler
CommandIntergration
LeaveCommand
PrivateChatRegistryTest
SelectCommand
UserRegistry

### CODE REVIEW

File: src/test/java/org/example/VChatTalk/command/impl/LeaveCommandTest.java [hunk 1]
```diff
@@ -28,51 +31,54 @@ class LeaveCommandTest {
...
     private LeaveCommand leaveCommand;
+    private final String SESSION_ID = "s1";
 
     @BeforeEach
     void setUp() {
         leaveCommand = new LeaveCommand(userRegistry, responder, privateChatRegistry);
+        lenient().when(session.getId()).thenReturn(SESSION_ID);
     }
 
     @Test
```
[the remark this answers is not in the document]
Comment (TQT279) [reply src/test/java/org/example/VChatTalk/command/impl/LeaveCommandTest.java:54]: LeaveCommand

File: src/test/java/org/example/VChatTalk/command/CommandIntegrationTest.java [hunk 2]
```diff
@@ -159,22 +171,4 @@ private void execute(String commandText, WebSocketSession session) throws IOExce
...
-        when(cmd2.getType()).thenReturn(CommandType.SELECT); // Duplicate!
-
-        List<IChatCommand> commands = Arrays.asList(cmd1, cmd2);
-
-        // Act & Assert: Expect the constructor to throw an IllegalStateException.
-        IllegalStateException exception = assertThrows(IllegalStateException.class, () -> {
-            new CommandExecutor(commands);
-        });
-    }
 }
```
[the remark this answers is not in the document]
Comment (TQT279) [reply src/test/java/org/example/VChatTalk/command/CommandIntegrationTest.java:198]: CommandIntergration

File: src/test/java/org/example/VChatTalk/command/impl/SelectCommandTest.java [hunk 3]
```diff
@@ -52,90 +49,42 @@ void testGetType() {
...
         CommandResult result = new CommandResult(CommandType.SELECT, "Alice", null);
         selectCommand.execute(session, result);
 
-        // Verify: The error ERR_NOT_LOGGED_IN must be submitted.
         verify(responder).sendError(session, MessageConstants.ERR_NOT_LOGGED_IN);
-        // Verify: Do not set targets
         verify(privateChatRegistry, never()).setTarget(anyString(), anyString());
     }
 
     @Test
```
[the remark this answers is not in the document]
Comment (TQT279) [reply src/test/java/org/example/VChatTalk/command/impl/SelectCommandTest.java:67]: SelectCommand

File: src/test/java/org/example/VChatTalk/handler/ChatWebSocketHandlerTest.java [hunk 4]
```diff
@@ -56,33 +54,24 @@ void afterConnectionEstablished_Success() throws Exception {
...
+        verify(userRegistry).removeUser(SESSION_ID); // QUAN TRỌNG: Verify handler gọi hàm xóa user
 
-        // Verify Follower Notification
-        verify(privateChatRegistry).removeTarget(followerId);
-        try {
-            verify(systemResponseSender).sendSystem(eq(followerSession), anyString());
-        } catch (Exception e) { /* ignored for test */ }
+        // Verify Broadcast
+        verify(broadcastService).broadcast(any(MessageDTO.class), eq(null));
     }
```
[the remark this answers is not in the document]
Comment (TQT279) [reply src/test/java/org/example/VChatTalk/handler/ChatWebSocketHandlerTest.java:77]: ChatWebSocketHandler

**Comment by TQT279** [comment]
PrivateChatRegistryTest
PrivateChatRegistryTest

**Comment by Toffu-BlankieChoccy** [comment]
Seems all good now. I'm merging this
